packages/flutter/lib/src/material/material.dart

```diff
@@ -147,6 +147,10 @@ class Material extends StatefulWidget {
   /// Creates a piece of material.
   ///
   /// The [type], [elevation] and [shadowColor] arguments must not be null.
+  ///
+  /// [shape] and [borderRadius] must not be both non-null.
+  /// If [type] is [MaterialType.circle] then [borderRadius] and [shape] must be
+  /// null.
```

Comment:
Another way to phrase these two additions would be "If a [shape] is specified, then the [borderRadius] property must not be null and the [type] property must not be [MaterialType.circle]. If the [borderRadius] is specified, then the [type] property must not be [MaterialType.circle]. In both cases, these restrictions are intended to catch likely errors."

Reply:
done